test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ESClientYamlSuiteTestCase.java

```diff
@@ -288,12 +290,23 @@ private static String buildSkipMessage(String description, SkipSection skipSecti
         return messageBuilder.toString();
     }
 
-    public void test() throws IOException {
+    public void test() throws IOException, ClientYamlTestParseException {
         //let's check that there is something to run, otherwise there might be a problem with the test section
         if (testCandidate.getTestSection().getExecutableSections().size() == 0) {
             throw new IllegalArgumentException("No executable sections loaded for [" + testCandidate.getTestPath() + "]");
         }
 
+        final DoSection templateDoSection = new DoSection();
+        final ApiCallSection templateApiCallSection = new ApiCallSection("indices.put_template");
+        templateApiCallSection.addParam("name", "global");
+        templateApiCallSection.addBody(Collections.singletonMap("template", "*"));
+        final Map<String, Integer> indexSettings = new HashMap<>();
+        indexSettings.put("number_of_shards", 1);
+        indexSettings.put("number_of_replicas", 0);
+        templateApiCallSection.addBody(Collections.singletonMap("settings", indexSettings));
+        templateDoSection.setApiCallSection(templateApiCallSection);
+        templateDoSection.execute(restTestExecutionContext);
```

Comment:
I see where the change comes from, but I am a bit on the fence on how we are solving the performance issue, cause we end up solving it only for the java runner, while the clients runners will still have slow tests. What is the overhead of relying on the default 5 shards compared to 1?